app/controllers/jobs_controller.rb

```diff
@@ -21,24 +34,43 @@ def create
   end
 
   def show
+    @company = Company.find(params[:company_id])
     @job = Job.find(params[:id])
+    @comment = Comment.new
+    @comment.job_id = @job.id
   end
 
   def edit
-    # implement on your own!
+    @company = Company.find(params[:company_id])
+    @job = Job.find(params[:id])
+    @categories = Category.all
   end
 
   def update
-    # implement on your own!
+    @company = Company.find(params[:company_id])
+    @job = Job.find(params[:id])
+    if @job.update(job_params)
+      flash[:success] = "#{@job.title} updated!"
+      redirect_to company_job_path(@company, @job)
+    else
+      @categories = Category.all
```

Comment:
`@categories` will come from the edit method and not through this so no need this piece isnt actually doing anything